### PR TITLE
Help Center: Add tailored section mapping

### PR DESCRIPTION
**Pull request by heavyweight**
This PR connects to the tailored articles backend changes done on D158372-code.
The tailored mapping are removed from the front-end.
To create a tailored section key that will match on the backend correctly a new mapping file was created: `tailored-section-mapping`. If the current route is connected to a tailored section it is sent to the backend, along with an empty query.

Fixes https://github.com/Automattic/dotcom-forge/issues/8502

## Proposed Changes

* Add tailored section mapping and send that to our endpoint

## Why are these changes being made?

* Keeping the article mapping on the front-end was preventing us from serving tailored articles across other locales without creating possible complex config.

## Testing Instructions
- Sync your sandbox with the latest trunk
- Checkout this branch and run calypso and also sync the help center app
- Test the tailored articles according to the tailored [sections defined](https://github.com/Automattic/wp-calypso/pull/93618/files#diff-3889c7f265a99203d6d76b224f47ac42dee677267aa23e16b333fd949c5ff04c)
- Test on different locales as well
- Ensure that the results returned are correct

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/help-center/src/tailored-section-mapping.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (arcangelini) [inline packages/help-center/src/tailored-section-mapping.json]: Do we need this file still? What happens if we just drop the beginning and ending slash and just send that with the query. Then rely on the API to decide if it exists or not.

**Review by arcangelini — APPROVED** [review]
Everything worked really well! Especially in places that we removed the `route-to-query` mappings.

One issue I did see was with Spanish pages. For some reason there is always one post missing. The english will return 5 but the spanish only gives 4 despite all the english pages having translations. I was able to replicate this when querying the API directly so it is unrelated to this PR.

**Review by escapemanuele — APPROVED** [review]
Tested together with D158992-code and it works great ✌️

**Comment by heavyweight** [comment]
With the patch D159242-code the tailored sections are ready to be mapped at the backend instead of here

**Comment by renancarvalho** [comment]
Looks good to me, tested and did not find anything wrong. 

Good job :)

**Pull request merged**